pubsub/google/cloud/pubsub_v1/subscriber/policy/base.py

```diff
@@ -31,6 +32,33 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+# Namedtuples for management requests. Used by the Message class to communicate
+# items of work back to the policy.
+AckRequest = collections.namedtuple(
+    'AckRequest',
+    ['ack_id', 'byte_size', 'time_to_ack'],
+)
+
+DropRequest = collections.namedtuple(
```

Comment:
One quibble I have with this is that `namedtuple` has a somewhat lousy equivalence method. In particular:

```python
DropRequest('ack123', 42) == LeaseRequest('ack123', 42)  # True
```

This is what data classes could solve if you did not need to support <= 3.5. ;-)

These are fine as-is, but be careful that on any tests you are always running `isinstance` checks. :-)

Reply:
I'm kind of okay with this equivalence - the only time this actually matters is in `dispatch_callback` which does a `type` switch.

Reply:
So long as the tests check `isinstance` I am okay with it.